### PR TITLE
Added ability to search books by title and author.

### PR DESCRIPTION
**Pull request by sovisou**

### CODE REVIEW

File: src/main/java/com/onlinebookstore/repository/book/TitleSpecificationProvider.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+import com.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TitleSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "title";
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/book/TitleSpecificationProvider.java]: create constant

File: src/main/java/com/onlinebookstore/repository/book/spec/AuthorSpecificationProvider.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    @Override
+    public String getKey() {
+        return "author";
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/book/spec/AuthorSpecificationProvider.java]: create constant

File: src/main/java/com/onlinebookstore/repository/book/spec/BookSpecificationBuilder.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParameters searchParameters) {
+        Specification<Book> specification = Specification.where(null);
+        if (searchParameters.titles() != null && searchParameters.titles().length > 0) {
+            specification = specification
+                    .and(bookSpecificationProviderManager.getSpecificationProvider("title")
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/book/spec/BookSpecificationBuilder.java]: create constant

File: src/main/java/com/onlinebookstore/repository/book/spec/BookSpecificationBuilder.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+                            .getSpecification(searchParameters.titles()));
+        }
+        if (searchParameters.authors() != null && searchParameters.authors().length > 0) {
+            specification = specification
+                    .and(bookSpecificationProviderManager.getSpecificationProvider("author")
```
Comment (Elena-Bruyako) [inline src/main/java/com/onlinebookstore/repository/book/spec/BookSpecificationBuilder.java]: create constant

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Good job! Let's fix some comments

File: src/main/java/com/onlinebookstore/repository/book/spec/AuthorSpecificationProvider.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+import com.onlinebookstore.model.Book;
+import com.onlinebookstore.repository.SpecificationProvider;
+import java.util.Arrays;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@Component
+public class AuthorSpecificationProvider implements SpecificationProvider<Book> {
+    private static final String AUTHOR_FIELD = "title";
```
Comment (ahoienko) [inline src/main/java/com/onlinebookstore/repository/book/spec/AuthorSpecificationProvider.java]:
```suggestion
    private static final String AUTHOR_FIELD = "author";
```

File: src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+import com.onlinebookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Component;
+
+@RequiredArgsConstructor
+@Component
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private static final String TITLE_FIELD = "title";
+    private static final String AUTHOR_FIELD = "title";
```
Comment (ahoienko) [inline src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java]:
```suggestion
    private static final String AUTHOR_FIELD = "author";
```

Comment (ahoienko) [inline src/main/java/com/onlinebookstore/repository/book/BookSpecificationBuilder.java]: This constants could be reused

**Review by ahoienko — CHANGES_REQUESTED** [review]
Please be cerafull and check PR before pushing to remote

**Review by okuzan — APPROVED** [review]

**Pull request merged**